pyannote/audio/core/pipeline.py

```diff
@@ -187,6 +182,133 @@ def from_pretrained(
+                Message to commit while pushing. Will default to `"Upload {object}"`.
+            private (`bool`, *optional*):
+                Whether or not the repository created should be private.
+            token (`bool` or `str`, *optional*):
+                The token to use as HTTP bearer authorization for remote files. If `True`, will use the token generated
+                when running `huggingface-cli login` (stored in `~/.huggingface`). Will default to `True` if `repo_url`
+                is not specified.
+            create_pr (`bool`, *optional*, defaults to `False`):
+                Whether or not to create a PR with the uploaded files or directly commit.
+            revision (`str`, *optional*):
+                Branch to push the uploaded files to.
+            commit_description (`str`, *optional*):
+                The description of the commit that will be created
+            tags (`List[str]`, *optional*):
+                List of tags to push on the Hub.
+            cache_dir: Path or str, optional
+                Path to config cache directory. Defauorch/pyannote" when unset.
+            config_yaml_path (`str`):
+                Path to the original "pyannote/speaker-diarization-3.1" Hub repository containing the default config file to use
+                to create custom speaker diarization config files.
+        """
+
+        api = HfApi()
+
+        _ = api.create_repo(
+            repo_id, private=private, token=token, exist_ok=True, repo_type="model"
+        )
+
+        # Load the pyannote/speaker-diarization-3.1 pipeline config:
+        try:
+            config_yml = hf_hub_download(
+                config_yaml_path,
+                "config.yaml",
+                repo_type="model",
+                revision=revision,
+                library_name="pyannote",
+                library_version=__version__,
+                cache_dir=cache_dir,
+                use_auth_token=token,
+            )
+        except RepositoryNotFoundError:
+            print(
+                f"""
+                Could not download '{config_yaml_path}' pipeline.
+                It might be because the pipeline is private or gated so make
+                sure to authenticate. Visit https://hf.co/settings/tokens to
+                create your access token and retry with:
+
+                >>> Pipeline.from_pretrained('{config_yaml_path}',
+                ...                          use_auth_token=YOUR_AUTH_TOKEN)
+
+                If this still does not work, it might be because the pipeline is gated:
+                visit https://hf.co/{config_yaml_path} to accept the user conditions."""
+            )
+            return None
+
+        with TemporaryDirectory() as tmpdir:
+
+            tmpdir = Path(tmpdir)
+
+            with open(config_yml, "r") as fp:
+                config = yaml.load(fp, Loader=yaml.SafeLoader)
+
+            if embedding_model is None:
```

Comment:
If you have a `.save_pretrained` method defined for the embedding/segmentation models, it becomes easy to save them locally to subfolders here